src/test/java/com/uid2/shared/store/reader/RotatingKeysetProviderTest.java

```diff
@@ -0,0 +1,140 @@
+package com.uid2.shared.store.reader;
+
+import com.uid2.shared.Utils;
+import com.uid2.shared.auth.AclSnapshot;
+import com.uid2.shared.auth.ClientKey;
+import com.uid2.shared.auth.Keyset;
+import com.uid2.shared.auth.KeysetSnapshot;
+import com.uid2.shared.cloud.ICloudStorage;
+import com.uid2.shared.cloud.InMemoryStorageMock;
+import com.uid2.shared.model.KeysetKey;
+import com.uid2.shared.store.CloudPath;
+import com.uid2.shared.store.scope.GlobalScope;
+import com.uid2.shared.store.scope.SiteScope;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import static com.uid2.shared.TestUtilites.makeInputStream;
+import static com.uid2.shared.TestUtilites.toInputStream;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+
+public class RotatingKeysetProviderTest {
+    private AutoCloseable mocks;
+    @Mock private ICloudStorage cloudStorage;
+    private RotatingKeysetProvider keysetProvider;
+
+    @Before
+    public void setup() {
+        mocks = MockitoAnnotations.openMocks(this);
+        keysetProvider = new RotatingKeysetProvider(cloudStorage, new GlobalScope(new CloudPath("metadata")));
+    }
+
+    private void addKeyset(JsonArray content, int siteId, int keysetId, int... listedSiteIds) {
+        JsonObject entry = new JsonObject();
+        entry.put("keyset_id", keysetId);
+        entry.put("site_id", siteId);
+        entry.put("name", "test");
+
+        JsonArray list = new JsonArray();
+        for (int listedSiteId : listedSiteIds) {
+            list.add(listedSiteId);
+        }
+        entry.put("allowed_sites", list);
+        entry.put("created", Instant.now().getEpochSecond());
+        entry.put("enabled", true);
+        entry.put("default", true);
+
+        content.add(entry);
+    }
+
+    private JsonObject makeMetadata(String location) {
+        JsonObject metadata = new JsonObject();
+        JsonObject keysets = new JsonObject();
+        keysets.put("location", location);
+        metadata.put("keysets", keysets);
+        return metadata;
+    }
+
+    @Test
+    public void loadsContentToSiteScope() throws Exception{
+        RotatingKeysetProvider provider = new RotatingKeysetProvider(cloudStorage, new SiteScope(new CloudPath("metadata"), 5));
+        JsonArray content = new JsonArray();
+        addKeyset(content, 5, 1, 1, 2, 3);
```

Comment:
these addKeysets are hard to read because of the list next to the other params. Maybe List.Of() would make them clearer?